### PR TITLE
0.0.10 - Adding support to auto-scaling configurations

### PR DESCRIPTION
**Pull request by thiagosp1**
# Pull Request

## What does this PR do?
It adds a simple version of auto-scaling, allowing the creation of an auto-scale policy using one of the predefined metrics, `RDSReaderAverageCPUUtilization` or `RDSReaderAverageDatabaseConnections`.
It is managed through a set of variables.

## Why is this PR being done?
To allow auto-scale configuration for clusters managed by this module.

## Checklist - These are **not** mandatory

- [x] I have updated the README.md if there are new procedures or changes.
- [x] I have updated CHANGELOG.md with new changes. 
- [x] I have deployed this change in another project successfully.

### CODE REVIEW

File: auto-scale.tf [hunk 1]
```diff
@@ -0,0 +1,25 @@
+resource "aws_appautoscaling_target" "aurora_scaling_target" {
+  count              = var.enable_auto_scale ? 1 : 0
+  service_namespace  = "rds"
+  resource_id        = "cluster:${aws_rds_cluster.primary.id}"
+  scalable_dimension = "rds:cluster:ReadReplicaCount"
```
Comment (ml-holdenporter) [inline auto-scale.tf:5]: Should we make this a variable with this value as the default?

File: auto-scale.tf [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  min_capacity       = var.auto_scale_min_capacity
+  max_capacity       = var.auto_scale_max_capacity
+}
+
+resource "aws_appautoscaling_policy" "aurora_scaling_policy" {
+  count              = var.enable_auto_scale ? 1 : 0
+  name               = "${var.auto_scale_metric} policy"
+  service_namespace  = "rds"
+  resource_id        = aws_appautoscaling_target.aurora_scaling_target[0].resource_id
+  scalable_dimension = "rds:cluster:ReadReplicaCount"
```
Comment (ml-holdenporter) [inline auto-scale.tf]: Same comment as above.

**Review by ml-holdenporter — CHANGES_REQUESTED** [review]

File: auto-scale.tf (hunk 1 above)
Comment (thiagosp1) [reply auto-scale.tf:5]: I don't think so man. The only acceptable value for scaling RDS is `cluster:ReadReplicaCount` 
So there is no real scenario where this value would have to be changed.
[Link to documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-applicationautoscaling-scalabletarget.html)
image

File: auto-scale.tf (hunk 2 above)
Comment (thiagosp1) [reply auto-scale.tf]: Changed to reference the value defined on the `aws_appautoscaling_target` resource

**Review by ml-holdenporter — APPROVED** [review]

**Pull request merged**